A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/PFD/Airbus_FMA.js

```diff
@@ -1000,45 +1004,44 @@ var Airbus_FMA;
         GetModeState_SRS() {
             if (Airbus_FMA.CurrentPlaneState.flightPhase == FlightPhase.FLIGHT_PHASE_TAKEOFF && Airbus_FMA.CurrentPlaneState.anyFlightDirectorsActive) {
                 if (Airbus_FMA.CurrentPlaneState.highestThrottleDetent == ThrottleMode.TOGA) {
-                    return Airbus_FMA.MODE_STATE.ENGAGED;
+                    SRSEnabled = true;
                 }
                 else if ((Airbus_FMA.CurrentPlaneState.highestThrottleDetent == ThrottleMode.FLEX_MCT) && (Airbus_FMA.CurrentPlaneState.flexTemperature > 0)) {
-                    return Airbus_FMA.MODE_STATE.ENGAGED;
-                }
-                else if ((SimVar.GetSimVarValue("L:AIRLINER_V2_SPEED", "Number") > 0) && (Airbus_FMA.CurrentPlaneState.flapsHandlePercent != 0)) {
-                    return Airbus_FMA.MODE_STATE.ARMED;
+                    SRSEnabled = true;
                 }
             }
             else if (Airbus_FMA.CurrentPlaneState.flightPhase == FlightPhase.FLIGHT_PHASE_GOAROUND) {
                 if ((Airbus_FMA.CurrentPlaneState.highestThrottleDetent == ThrottleMode.TOGA) &&
                     (Airbus_FMA.CurrentPlaneState.flapsHandlePercent != 0)) {
-                    return Airbus_FMA.MODE_STATE.ENGAGED;
+                        SRSEnabled = true;
+                        return Airbus_FMA.MODE_STATE.ENGAGED;
                 }
             }
+            if (Airbus_FMA.CurrentPlaneState.flightPhase != FlightPhase.FLIGHT_PHASE_TAKEOFF) {
+                SRSEnabled = false;
+            }
+            if (SRSEnabled) {
+                return Airbus_FMA.MODE_STATE.ENGAGED;
+            }
             return Airbus_FMA.MODE_STATE.NONE;
         }
         static GetModeState_ALT() {
             if (Airbus_FMA.CurrentPlaneState.anyAutoPilotsActive && Airbus_FMA.CurrentPlaneState.autoPilotAltitudeLockActive && Airbus_FMA.CurrentPlaneState.anyFlightDirectorsActive) {
-                if (Airbus_FMA.CurrentPlaneState.autoPilotAltitudeArmed) {
-                    return Airbus_FMA.MODE_STATE.ARMED;
+                var diffAlt = Math.abs(Airbus_FMA.CurrentPlaneState.autoPilotAltitudeLockValue - Airbus_FMA.CurrentPlaneState.altitude);
+                if (diffAlt <= Airbus_FMA.Definitions.ALT_ENGAGED_RANGE) {
+                    this._lastALTMode = Airbus_FMA.MODE_STATE.ENGAGED;
                 }
```

Comment:
This doesn't seem right.

You are starting an if statement `if (Airbus_FMA.CurrentPlaneState.autoPilotAltitudeArmed) {` and returning immediately, but you add more statements after. Those statements will never be executed since you return.

You probably meant this : 

```
if (Airbus_FMA.CurrentPlaneState.autoPilotAltitudeArmed) {
    return Airbus_FMA.MODE_STATE.ARMED;
}
```

Reply:
Checking now, it was a long day! Spent maybe 8 hours straight working on this stuff, I was expecting some errors somewhere! It worked fine in the sim so I'll have a look what I messed up. 👍

Reply:
For the above, I believe it's just displaying weirdly in the changes? In the file itself it looks correct and doesn't contain the if statement shown above. On top of that, the file has no changed compared to the repo.

Reply:
Nevermind, I was tired and for some reason forgot that red in the diff means removed 😂